homeassistant/components/zwave_js/device_trigger.py

```diff
@@ -141,10 +185,95 @@
     BASIC_VALUE_NOTIFICATION_SCHEMA,
     CENTRAL_SCENE_VALUE_NOTIFICATION_SCHEMA,
     SCENE_ACTIVATION_VALUE_NOTIFICATION_SCHEMA,
+    CONFIG_PARAMETER_VALUE_UPDATE_SCHEMA,
+    VALUE_VALUE_UPDATE_SCHEMA,
     NODE_STATUS_SCHEMA,
 )
 
 
+@dataclass
+class ValueUpdatesAutomationTracker:
+    """Class to store an unsubscription callback for a given automation."""
+
+    value_tracker: ValueUpdatesValueTracker
+    id: str
+    unsub: Callable
+
+
+@dataclass
+class ValueUpdatesValueTracker:
+    """Class to store tracked automations for a given value."""
+
+    device_tracker: ValueUpdatesDeviceTracker
+    zwave_value: Value
+    prev_value: Any
+    automations: dict[str, ValueUpdatesAutomationTracker] = field(
+        default_factory=dict, init=False
+    )
+
+
+@dataclass
+class ValueUpdatesDeviceTracker:
+    """Class to store tracked values for a given device/node."""
+
+    hass: HomeAssistant
+    config_entry_id: str
+    device_id: str
+    node: Node
+    value_update_unsub: Callable = field(init=False)
+    values: dict[str, ValueUpdatesValueTracker] = field(
+        default_factory=dict, init=False
+    )
+
+    def __post_init__(self) -> None:
+        """Handle post initialization."""
+        self.value_update_unsub = self.node.on(
+            EVENT_VALUE_UPDATED,
+            lambda event: self.fire_value_updated_event(event["value"]),
+        )
+
+    @callback
+    def fire_value_updated_event(self, value: Value) -> None:
+        """Fire a value updated event if needed."""
+        # If we are not tracking this value, we can skip the rest of the logic
+        if (value_tracker := self.values.get(value.value_id)) is None:
+            return
+
+        # In order to support from states, we need to store the previous value
+        # of the value in the value tracker
+        from_value = value_tracker.prev_value
+        to_value = value_tracker.prev_value = value.value
+        event_data = {
+            ATTR_DEVICE_ID: self.device_id,
+            ATTR_COMMAND_CLASS: value.command_class,
+            ATTR_PROPERTY: value.property_,
+            ATTR_PROPERTY_KEY: value.property_key,
+            ATTR_ENDPOINT: value.endpoint,
+            state.CONF_FROM: from_value,
+            state.CONF_TO: to_value,
+        }
+        LOGGER.debug(
+            "Firing device trigger event: %s",
+            {k: v for k, v in event_data.items() if v is not None},
+        )
+        self.hass.bus.async_fire(
+            ZWAVE_JS_DEVICE_TRIGGER_VALUE_UPDATED_EVENT,
+            {k: v for k, v in event_data.items() if v is not None},
+        )
+
+
+async def async_validate_trigger_config(
```

Comment:
@kpine will hopefully post more details, but when testing this he discovered that on startup, this step raises an exception. It seems that this validation logic is run before the `zwave_js` integration is fully setup which means we can't get the node because the integration hasn't initialized state yet. I *think* this is an upstream problem with the automation logic, but looking for feedback on this

Reply:
Stack trace of the exception: https://paste.ubuntu.com/p/bNTq7Nn4my/